### PR TITLE
clio-30006 have thTable automatically fix incorrect currentPage values

### PR DESCRIPTION
**Pull request by andreisoare**

### CODE REVIEW

File: themis_components/thAlert/thAlertAnchor.directive.spec.coffee [hunk 1]
```diff
@@ -10,15 +10,24 @@ describe "ThemisComponents: Directive: thAlertAnchor", ->
   describe "load message data", ->
     it 'should load the success data', ->
       AlertManager.showSuccess "Risus Lorem"
-      expect(scope.alertAnchor.alertMessage).toEqual {message: 'Risus Lorem', type: 'success'}
+      expect(scope.alertAnchor.alertMessage).toEqual {
```
Comment (andreisoare) [inline themis_components/thAlert/thAlertAnchor.directive.spec.coffee]: Lines too long.

File: themis_components/thTable/thTablePagination.service.coffee [hunk 2]
```diff
@@ -105,6 +109,11 @@ angular.module 'ThemisComponents'
       updatePagination: (options) ->
         totalItems = options.totalItems ? totalItems
         currentPage = options.currentPage ? currentPage
+        if currentPage < 1
+          currentPage = 1
```
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l113)

File: themis_components/thTable/thTablePagination.service.coffee [hunk 3]
```diff
@@ -59,18 +61,20 @@ angular.module 'ThemisComponents'
...
       goToPage: (page) ->
-        return if page is ellipsis or \
-                  page not in [1 .. totalPages()] or \
-                  page is currentPage
+        return if page in [ellipsis, currentPage]
+
+        if page < 1
+          page = 1
+        else if totalPages() isnt 0 and page > totalPages()
+          page = totalPages()
```
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l74)

File: themis_components/thTable/thTablePagination.service.coffee [hunk 4]
```diff
@@ -59,18 +61,20 @@ angular.module 'ThemisComponents'
...
-      goToPrevPage: ->
-        if currentPage > 1
-          self.goToPage currentPage - 1
+      goToPrevPage: -> self.goToPage currentPage - 1
 
       goToPage: (page) ->
-        return if page is ellipsis or \
-                  page not in [1 .. totalPages()] or \
-                  page is currentPage
+        return if page in [ellipsis, currentPage]
```
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l69)

File: themis_components/thTable/thTablePagination.service.coffee [hunk 5]
```diff
@@ -59,18 +61,20 @@ angular.module 'ThemisComponents'
...
+
+        if page < 1
+          page = 1
```
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l72)

File: themis_components/thTable/thTablePagination.service.coffee [hunk 6]
```diff
@@ -105,6 +109,11 @@ angular.module 'ThemisComponents'
       updatePagination: (options) ->
         totalItems = options.totalItems ? totalItems
         currentPage = options.currentPage ? currentPage
+        if currentPage < 1
+          currentPage = 1
+        else if totalPages() isnt 0 and currentPage > totalPages()
+          currentPage = totalPages()
```
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l115)

File: themis_components/thTable/thTableDelegate.service.spec.coffee [hunk 7]
```diff
@@ -50,6 +50,25 @@ describe 'ThemisComponents: Service: thTableDelegate', ->
...
+      pageSize = 5
+      numPages = 10
+      totalItems = pageSize * numPages
+      currentPageFromFetchData = undefined
+      headers = [
+        TableHeader {sortField: ''}
+        TableHeader {sortField: '', sortActive: true}
+      ]
+      fetchData = (options, updateData) ->
+        timesCalled += 1
```
Comment (seanhealy) [inline themis_components/thTable/thTableDelegate.service.spec.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901448/issues/source?bid=3086685&fileBranchId=3086932#l65)

File: themis_components/thTable/thTableDelegate.service.spec.coffee [hunk 8]
```diff
@@ -50,6 +50,25 @@ describe 'ThemisComponents: Service: thTableDelegate', ->
       delegate = TableDelegate {fetchData, headers}
       expect(called).toBe true
 
+    it 'calls fetchData again with last page if currentPage > last page', ->
+      timesCalled = 0
```
Comment (seanhealy) [inline themis_components/thTable/thTableDelegate.service.spec.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901448/issues/source?bid=3086685&fileBranchId=3086932#l54)

File: themis_components/thTable/thTablePagination.service.coffee (hunk 2 above)
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l113)

File: themis_components/thTable/thTablePagination.service.coffee (hunk 3 above)
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l74)

File: themis_components/thTable/thTablePagination.service.coffee (hunk 4 above)
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l69)

File: themis_components/thTable/thTablePagination.service.coffee (hunk 5 above)
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l72)

File: themis_components/thTable/thTablePagination.service.coffee (hunk 6 above)
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l115)

File: themis_components/thTable/thTableDelegate.service.spec.coffee (hunk 7 above)
Comment (seanhealy) [inline themis_components/thTable/thTableDelegate.service.spec.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901448/issues/source?bid=3086685&fileBranchId=3086932#l65)

File: themis_components/thTable/thTableDelegate.service.spec.coffee (hunk 8 above)
Comment (seanhealy) [inline themis_components/thTable/thTableDelegate.service.spec.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901448/issues/source?bid=3086685&fileBranchId=3086932#l54)

File: themis_components/thTable/thTablePagination.service.coffee (hunk 2 above)
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l113)

File: themis_components/thTable/thTablePagination.service.coffee (hunk 3 above)
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l74)

File: themis_components/thTable/thTablePagination.service.coffee (hunk 4 above)
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l69)

File: themis_components/thTable/thTablePagination.service.coffee (hunk 5 above)
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l72)

File: themis_components/thTable/thTablePagination.service.coffee (hunk 6 above)
Comment (seanhealy) [inline themis_components/thTable/thTablePagination.service.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901451/issues/source?bid=3086685&fileBranchId=3086932#l115)

File: themis_components/thTable/thTableDelegate.service.spec.coffee (hunk 7 above)
Comment (seanhealy) [inline themis_components/thTable/thTableDelegate.service.spec.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901448/issues/source?bid=3086685&fileBranchId=3086932#l65)

File: themis_components/thTable/thTableDelegate.service.spec.coffee (hunk 8 above)
Comment (seanhealy) [inline themis_components/thTable/thTableDelegate.service.spec.coffee]: Codacy Issue found: [Line contains inconsistent indentation](https://www.codacy.com/app/s_2/ThemisUI/file/1862901448/issues/source?bid=3086685&fileBranchId=3086932#l54)

**Pull request merged**